src/core/expression/predicate.visitor.spec.js

```diff
@@ -0,0 +1,239 @@
+import {PredicateVisitor} from './predicate.visitor';
+
+function valueFactory() {
+	return (value) => value;
+}
+describe('PredicateVisitor', () => {
+
+	describe('visitGroup', () => {
+		it('check for equality with op = "and"', () => {
+			let group = {
+				kind: 'group',
+				op: 'and',
+				right: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				},
+				left: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				}
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(group);
+			let res = foo(123);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "or"', () => {
+			let group = {
+				kind: 'group',
+				op: 'or',
+				right: {
+					kind: 'condition',
+					op: 'equals',
+					right: 'Some string',
+					left: 'value'
+				},
+				left: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				}
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(group);
+			let res = foo(123);
+			expect(res).to.equal(true);
+			expect(foo('Some string')).to.equal(true)
+		});
+	});
+
+	describe('visit condition', () => {
+		it('check for equality with op = "notEquals"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'notEquals',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(999);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "greaterThan"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'greaterThan',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(999);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "lessThan"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'lessThan',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(1);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "between"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'between',
+				right: [10, 20],
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(15);
+			expect(res).to.equal(true);
```

Comment:
can you make separate test for each case? I mean one theory contains one expectation
smth like
```it('check equality with op = "between" inside range')```
```it('check equality with op = "between" on bounds of range ')```
```it('check equality with op = "between" outside range')```